exporter/opentelemetry-exporter-otlp/src/opentelemetry/exporter/otlp/exporter.py

```diff
@@ -201,15 +201,23 @@ def __init__(
             self._client = self._stub(
                 insecure_channel(endpoint, compression=compression_algorithm)
             )
-        else:
-            credentials = credentials or _load_credential_from_file(
-                Configuration().EXPORTER_OTLP_CERTIFICATE
-            )
-            self._client = self._stub(
-                secure_channel(
-                    endpoint, credentials, compression=compression_algorithm
-                )
+            return
+
+        # secure mode
+        if (
+            credentials is None
+            and Configuration().EXPORTER_OTLP_CERTIFICATE is None
+        ):
+            raise ValueError("No credentials set in secure mode.")
```

Comment:
any advantage to raising an exception here and not default to using `grpc.ssl_channel_credentials()`?

Reply:
This case is when insecure=False which is by default. Without this custom error handling it would break deep in stack inside grpc library without a useful message. Or can we handle this in a better way?

Reply:
It's probably reasonable. I don't know how bad the original error is but asserting invalid configurations is certainly helpful.

Reply:
@jan25 my suggestion here was that using `ssl_channel_credentials` without any parameters would work in the case of a secure connection as it will load the root certificate from a default location:
https://grpc.github.io/grpc/python/grpc.html#create-client-credentials

Reply:
I like this idea, perhaps we need to update the spec before using default location? Also, i was wrong about grpc library erroring. Actually it was `open(filename)` thats erroring when no cert location is set and `filename=None`.

Reply:
@jan25 if it violates the spec to look for ssl credentials in the secure case before failing, then agreed we should amend the spec.

Can you file an issue? Or do you want someone else to handle that conversation?

Reply:
Hi @toumorokoshi I recently learnt grpc python library has the default certificate path support, so i don't have 100% context of this. Could someone else open an issue in spec repo about this?